com.unity.render-pipelines.high-definition/Runtime/Lighting/Light/HDGpuLightsBuilder.Jobs.cs

```diff
@@ -0,0 +1,748 @@
+using System;
+using UnityEngine.Jobs;
+using Unity.Jobs;
+using Unity.Collections;
+using Unity.Collections.LowLevel.Unsafe;
+using System.Threading;
+
+namespace UnityEngine.Rendering.HighDefinition
+{
+    internal partial class HDGpuLightsBuilder
+    {
+        JobHandle m_CreateGpuLightDataJobHandle;
+
+        internal struct CreateGpuLightDataJobGlobalConfig
+        {
+            public bool lightLayersEnabled;
+            public float specularGlobalDimmer;
+            public int invalidScreenSpaceShadowIndex;
+            public float maxShadowFadeDistance;
+
+            public static CreateGpuLightDataJobGlobalConfig Create(
+                HDCamera hdCamera,
+                HDShadowSettings hdShadowSettings)
+            {
+                return new CreateGpuLightDataJobGlobalConfig()
+                {
+                    lightLayersEnabled = hdCamera.frameSettings.IsEnabled(FrameSettingsField.LightLayers),
+                    specularGlobalDimmer = hdCamera.frameSettings.specularGlobalDimmer,
+                    maxShadowFadeDistance = hdShadowSettings.maxShadowDistance.value,
+                    invalidScreenSpaceShadowIndex = (int)LightDefinitions.s_InvalidScreenSpaceShadow
+                };
+            }
+        }
+
+#if ENABLE_BURST_1_5_0_OR_NEWER
+        [Unity.Burst.BurstCompile]
+#endif
+        internal struct CreateGpuLightDataJob : IJobParallelFor
+        {
+            #region Parameters
+            [ReadOnly]
+            public int totalLightCounts;
+            [ReadOnly]
+            public int outputLightCounts;
+            [ReadOnly]
+            public int outputDirectionalLightCounts;
+            [ReadOnly]
+            public int outputLightBoundsCount;
+            [ReadOnly]
+            public CreateGpuLightDataJobGlobalConfig globalConfig;
+            [ReadOnly]
+            public Vector3 cameraPos;
+            [ReadOnly]
+            public int directionalSortedLightCounts;
+            [ReadOnly]
+            public bool isPbrSkyActive;
+            [ReadOnly]
+            public int precomputedAtmosphericAttenuation;
+            [ReadOnly]
+            public int defaultDataIndex;
+            [ReadOnly]
+            public int viewCounts;
+            [ReadOnly]
+            public bool useCameraRelativePosition;
+
+            //sky settings
+            [ReadOnly]
+            public Vector3 planetCenterPosition;
+            [ReadOnly]
+            public float planetaryRadius;
+            [ReadOnly]
+            public float airScaleHeight;
+            [ReadOnly]
+            public float aerosolScaleHeight;
+            [ReadOnly]
+            public Vector3 airExtinctionCoefficient;
+            [ReadOnly]
+            public float aerosolExtinctionCoefficient;
+            #endregion
+
+            #region input light entity SoA data
```

Comment:
This is a AoS though? Is the region name a sort-of TODO or I am missing something?

Reply:
Renamed this.